libtiff/libtiff_ctypes.py

```diff
@@ -85,6 +85,8 @@
     if not os.path.isfile(include_tiff_h):
         include_tiff_h = os.path.join(os.path.split(lib)[0], 'include',
                                       'tiff.h')
+    if not os.path.isfile(include_tiff_h):
+        include_tiff_h = os.path.join('/usr', 'local', 'include', 'tiff.h')
```

Comment:
If the header file is in `/usr/local/include` then where is the libtiff library itself installed in your system? Or to put it differently, what is `lib` in your case?

Reply:
I built 4.5.0 locally and used `--prefix=/usr/local` as part of the configure parameters, so it is at /usr/local/lib

Reply:
Maybe it would be better to use something to parse `pkg-config libtiff-4 --cflags-only-I`.

Reply:
In this situations I suggest using
```suggestion
    if not os.path.isfile(include_tiff_h):
        include_tiff_h = os.path.join(os.path.dirname(lib), '..', 'include', 'tiff.h')
```
that applies to all prefix cases where one has `<prefix>/include/tiff.h` and `<prefix>/lib/libtiff.so` (or similar).

Reply:
There is already a line that tries that.  The problem is that (on linux at least), `ctypes.util.find_library('tiff')` returns something without a path (e.g., `libtiff.so.6`), so basing the path from that doesn't work.  Internally, that find_library command often uses `ldconfig -p` and searches for the library which can list the path, but the return value doesn't (see https://github.com/python/cpython/blob/2.7/Lib/ctypes/util.py#L261).

Reply:
Right. So, there is not much we can do but hardcode prefixes like `/usr/local`. I would like to avoid such hardcoding as there exists a not-a-small number of custom prefixes (`/opt`, `/opt/local`, etc) that people may use, and supporting these is a pain.

How about introducing an environment variable, say `LIBTIFF_HOME` or `LIBTIFF_PREFIX`, that pylibtiff will use for finding the libtiff header when specified? In this way, we are able to support all custom locations of libtiff headers/libraries.
What do you think?

Reply:
Sure -- an environment variable would work.  I think it should be the path to the include file (e.g., in my case /usr/local/include), so maybe a name of `LIBTIFF_INCLUDE_DIR` would be clearer?  Or we could specify the full path to the header file (e.g., /usr/local/include/tiff.h), in which case `LIBTIFF_INCLUDE_FILEPATH` might be clearer?

Reply:
I feel that `INCLUDE` part is superfluous if we specify the path to the header file and not to the path of include directory. 
How about `TIFF_HEADER_FILEPATH` or `TIFF_HEADER_PATH`?